### PR TITLE
16.0.1.1 flight phase views constraint

### PR DESCRIPTION
**Pull request by annzzu**
[IMP] flight_event: add phase duration views and constraints

- Add tree, form, and search views and menu item for flight.phase.duration model
- Implement constraint to ensure correct event time sequence per phase/flight
- Update README

16.0.1.1.0 version

### CODE REVIEW

**Comment by ayushin** [comment]
can we think of /consider a solution to calculate durations programmatically without storing them in the database model?

**Comment by ayushin** [comment]
further, we should not be hard-coding "A" for only calculating actual durations

**Comment by annzzu** [comment]
I have already added the `time_kind` field in `flight.phase.duration`, which allows us to retrieve durations filtered by both phase and time kind.

```python
def get_duration(self, phase_name, time_kind):
    """
    Get the duration for a specific phase and time kind.
    Returns the duration or 0 if not found.
    """
    pd_id = self.phase_duration_ids.filtered(lambda pd_id:
                                             pd_id.time_kind == time_kind
                                             and pd_id.phase_id.name == phase_name)

    return pd_id.duration or 0
```

Now, there are no more hardcoded phases or time kinds. You can simply specify the phase and time kind you want, and the corresponding duration will be returned. even better solution than block_a

File: flight_event/models/flight_flight.py [hunk 1]
```diff
@@ -29,37 +41,94 @@ def _check_event_sequence(self):
...
+        # Search for all existing phase durations for this flight
+        existing_durations = PhaseDuration.search([('flight_id', '=', self.id)])
+
+        # Unlink any duration records that were not updated
+        for duration in existing_durations:
+            duration_info = (duration.phase_id.id, duration.time_kind)
+            if duration_info not in updated_phase_durations:
+                duration.unlink()
+
+    def get_duration(self, phase_name, time_kind):
```
Comment (annzzu) [inline flight_event/models/flight_flight.py:113]: function which calculates duration with phase and time kind

**Comment by ayushin** [comment]
please rebase on the current branch so i can review at once

**Comment by annzzu** [comment]
This is rebased. you can check

**Comment by ayushin** [comment]
closed in favour of complete solution

**Pull request closed without merging**